tests/atomese/atom_types/newlink.scm

```diff
@@ -0,0 +1,8 @@
+(use-modules (opencog) (opencog exec) (opencog asmoses))
+
+(define test_or
+(New (TypeNode "OrLink") (ListLink (Predicate "P") (Predicate "Q"))))
+
+(define test_and
+(New (TypeNode "AndLink") (ListLink (Predicate "P") (Predicate "Q"))))
```

Comment:
So you've settled with the format
```
(NewLink (TypeNode <atomtype>) (List <outgoing-1> ... <outgoing-n>))
```
Another possible format would be to flatten the list of outgoings, so
```
(NewLink (TypeNode <atomtype>) <outgoing-1> ... <outgoing-n>)
```
I can't tell what is best. The advantage of the first format is that re-occuring Lists would point to the same subhypergraphs in the atomspace. The advantage of the second format is that it is closer to the existing scheme function `cog-new-link`. Maybe both could be supported? Though it would have the problem that if one wants to wrap a certain `List` in a link, it would make the `List` disappear, and so the user would need to escape the first list with a second list, which I don't like very much (links should be as simple and have the least amount of special cases).

What do you guys (and girls, to not be sexist, or should I call everybody girls for simplicity?) think?

Hmm, since it is a rather core atom type, maybe we should move that work to the atomspace repo...